### ISSUE TITLE
Nick's TODO's (can also be done by community)

### ISSUE THREAD

**Issue by baughmann** [seven-of-eleven/designare-z390-opencore-efi#57]
This is my TODO list for this repo. I've been really slacking on updates since I'm mostly using Ubuntu these days. That said, I want to keep this repo up-to-date in case I want to switch back to macOS full time.

1. Update OC to latest (I believe macOS still works on current version)
2. Post new BIOS config for F9
    - Optionally update some screenshots for F9
3. Make it known that I've stopped using the Fenvi because it's janky AF
4. ???
5. Profit

**Comment by jahwookie** [comment]
I used Ubuntu and a couple of other Linux distros for years. Incredibly
stable... but none of my audio software runs on it, except in a
self-defeating way through Wine. Well, I take that back: Presonus does
maintain a Linux version of their Studio One DAW, and there are a couple of
native DAWs that have gotten better. But virtually none of the hundreds of
plug-ins I use will work in that environment. So, Mac/Hackintosh is the way
to go for me at least for now (along with a WIndows 10 laptop).

All that said, it seems to me that some kind of enhanced virtualization is
the post-Intel future for Hackintosh – if, indeed, there is a future.
Wondering what anyone else's take is on this. JW

**Comment by baughmann** [comment]
@jahwookie Im not too sure there is a long term future. After Apple drops support for Intel Macs, I’d guess that it would be quite a long time before hackintoshes with good performance re-emerge. Like you stated, the future probably lies with virtualization

**Comment by jahwookie** [comment]
Maybe so, and that virtualization is easily done in Linux! So maybe I'll be
able to come back to Ubuntu one day...

**Comment by jordanurbs** [comment]
CaseySJ added a way to use the internal BT/WiFI in the motherboard:

https://www.tonymacx86.com/threads/asus-z690-proart-creator-wifi-thunderbolt-4-i7-12700k-amd-rx-6800-xt.318311/post-2331654

Any thoughts to integrate that into the repo? I know it means losing a USB port... or two...

**Comment by seven-of-eleven** [comment]
@jordanurbs Thanks for letting me know about this. There should be a way to include it, maybe as an alternate EFI.  Looks like they are using is SSDT patching to enable disable USB ports. Leave it with me for the next release.

Thanks again.

**Comment by jordanurbs** [comment]
> @jordanurbs Thanks for letting me know about this. There should be a way to include it, maybe as an alternate EFI. Looks like they are using is SSDT patching to enable disable USB ports. Leave it with me for the next release.
> 
> Thanks again.

Cheers @seven-of-eleven! And thanks for taking over this repo.

Just some feedback. CaseySJ also released an EFI set-up using internal BT/WiFi and iMacPro19,1, which is what I was using for a month or two, up until this week. It has HS14 enabled but none of the USB-C/thunderbolt ports work unless they are true thunderbolt devices (otherwise they are disabled as USB devices)--which can be frustrating.

So I decided to give this repo a try again since you've taken over in the legend @baughmann's absence, and it still gives me one USB-C port behind the machine.

However..... On this EFI (same hardware), I get random shutdowns (mostly during sleep) and it seems overall poorer performance with this compared to CaseySJ's new-ish EFI. I haven't run any actual tests to confirm poorer performance but just feels that way.

I have been trying myself to disable HS11 and enable HS14 for mobo BT/Wifi to see how it'll perform in this EFI, but with no luck, because I still haven't gotten the hang of mapping USB Ports. CaseySJ also kept it at iMac19,1, which also may have something to do with the issues I'm experiencing using MacPro7,1

I would personally love to try a version of this repo enable internal BT/WiFi while leaving a USB-C port open, and to see how it fares with iMac19,1

Cheers!

**Comment by seven-of-eleven** [comment]
@jordanurbs I'm working on an update now that OC 0.8.5 is released. It should be complete in a couple of days.

**Comment by seven-of-eleven** [comment]
@jahwookie unfortunately I don't have a Thunderbolt device to test with. If you can recommend a cheap Thunderbolt device I'll pick it up and test it out. Thanks.

**Comment by seven-of-eleven** [comment]
@jordanurbs let me know if the 2nd EFI works for you. It is working for me. All the best.

**Comment by jordanurbs** [comment]
@seven-of-eleven THANK YOU!!! 

I feel like I can breathe again. Machine is running incredibly, and I have so many ports!

Your efforts are appreciated :)

**Comment by JonMcL** [comment]
@jordanurbs are you on Monterey or Big Sur? And if Monterey, is Bluetooth 100% reliable for you?

I had previously used OpenIntelWireless kexts for bluetooth with Big Sur and had mostly success. Not so much with Monterey. Weird "volume hash mismatch" errors and bluetooth connection instability. 

Some examples:
- AirPod Pros would sometimes just cut out and it seemed like Bluetooth completed crashed. Had to restart.
- Trackpad would never reliably connect after a restart. Sometimes couldn't get it to connect again. Battery was draining VERY quickly (suggesting disconnecting over and over again)

My original motivation for using OpenIntelWireless for Bluetooth was to eliminate the need for the internal motherboard USB 2.0 header as part of the USB port mapping for the Fenvi card. My Bluetooth troubles _might_ have been caused by using a mixture -- Bluetooth from the Intel chipset and WiFi from the Fenvi/Broadcom card.

So, for me personally, now that I'm on Monterey, I've switched backto Fenvi/Broadcom for both Bluetooth and WiFi and I've disabled the Intel Wifi option in the BIOS. Since I have the Fenvi card, I'm not really sure why I was fighting it :)  I may give it a try again in a few months if it looks like there is more progress at OpenIntelWireless.

**Comment by seven-of-eleven** [comment]
@JonMcL I'm on Monterey and using the internal Bluetooth now as well. I'm not having any issues with my Apple TrackPad 2. I leave it on continuously and it connects automatically when I startup or reboot. I use my Sony xm4s with it as well without issue, but typically only for 1-2 hours at a time.

 **I don't use the internal WiFi though** (it works I just turn it off). I used to have a PCIE wifi card (don't remember the brand) and I alway experienced drops between WiFi and Bluetooth as you've described. Looking into it I was told that it's b/c they both use the 2.4 GHz range. Some folks suggested connecting WiFi to 5GHz to avoid this. I never tried that, I just turned it off, as I prefer to use ethernet.

Anyways thought I'd share my experience as well. If your system is working, I wouldn't rush to change it. If you do be sure to take a backup of your EFI folder. Best of luck.
